crowbar_framework/lib/crowbar/lock/local_blocking.rb

```diff
@@ -17,21 +17,18 @@
 
 module Crowbar
   class Lock::LocalBlocking < Lock
-    def acquire
+    def acquire(options = {})
       logger.debug("Acquire #{name} lock enter as uid #{Process.uid}")
-      begin
-        @file ||= File.new(path, File::RDWR | File::CREAT, 0644)
-      rescue
-        logger.error("Couldn't open #{path} for locking: #$!")
-        logger.error("cwd was #{Dir.getwd})")
-        raise "Couldn't open #{path} for locking: #$!"
-      end
-      logger.debug("Acquiring #{name} lock")
+      ensure_lock_file_exists
+      logger.debug("Acquiring #{name} lock with #{options}")
       count = 0
+      # specify different bits for shared vs. exclusive locks
+      bits = File::LOCK_NB
+      bits |= options[:shared] ? File::LOCK_SH : File::LOCK_EX
```

Comment:
With `options[:shared]` being nil, the last part of ternary operator is used, so yes, it works.